pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/impl/UserService.java

```diff
@@ -580,4 +590,130 @@ public User createOpenAccessUser(Role openAccessRole) {
         return user;
     }
 
+    /**
+     * Using the introspection token response, load the user from the database. If the user does not exist, we
+     * will reject their login attempt. For the RAS integration here is a sample payload.
+     *
+     * @param node The response from the introspect endpoint
+     * @return The user
+     */
+    public Optional<User> createRasUser(JsonNode node, Connection connection) {
+        try {
+            String userEmail = node.get("preferred_username").asText();
+            logger.info("Loading user for sub: {}", userEmail);
+
+            User new_user = new User();
+            new_user.setSubject(connection.getSubPrefix() + userEmail);
+            new_user.setEmail(userEmail);
+            new_user.setConnection(connection);
+            User actual_user = this.findOrCreate(new_user);
+
+            if (actual_user.getRoles() == null) {
+                actual_user.setRoles(new HashSet<>());
+            }
+
+            actual_user.setAcceptedTOS(new Date());
+            logger.info("LOGIN SUCCESS ___ USER DATA: {}", actual_user);
+            return Optional.of(actual_user);
+        } catch (Exception e) {
+            logger.error("Failed to create user from introspect response", e);
+            return Optional.empty();
+        }
+    }
+
+    public Set<User> getAllUsersWithAPassport() {
+        return this.userRepository.findByPassportIsNotNull();
+    }
+
+    /**
+     * Clears users session and merge template which effectively logs them out.
+     *
+     * @param user
+     */
+    public void logoutUser(User user) {
+        evictFromCache(user.getSubject());
+        this.removeUserPassport(user.getSubject());
+        this.sessionService.endSession(user.getSubject());
+    }
+
+    /**
+     * Update the provided users roles based on the list of roleNames provided. This method will update the roles
+     * in place. Adding and removing roles from the current list of roles.
+     *
+     * @param current_user User to be updated
+     * @param roleNames    Roles that should be assigned to the user.
+     */
+    public User updateUserRoles(User current_user, Set<String> roleNames) {
```

Comment:
Where are these underscored names coming from? I know we're lax with linting, but there are _some_ conventions. Camel case please.

Reply:
I believe it was because of the refactor. I did a method extract with intellij. I think it keeps the existing names.